sdk/tests/js/tests/tex-image-and-sub-image-2d-with-image-bitmap-utils.js

```diff
@@ -0,0 +1,169 @@
+        loc = gl.getUniformLocation(program, "face");
+    }
+
+    var tolerance = 10;
+    for (var tt = 0; tt < targets.length; ++tt) {
+        if (bindingTarget == gl.TEXTURE_CUBE_MAP) {
+            gl.uniform1i(loc, targets[tt]);
+        }
+        // Draw the triangles
+        wtu.clearAndDrawUnitQuad(gl, [0, 0, 0, 255]);
+
+        // Check the top pixel and bottom pixel and make sure they have
+        // the right color.
+        debug("Checking " + (flipY ? "top" : "bottom"));
+        wtu.checkCanvasRect(gl, quaterWidth, bottom, 2, 2, tl, "shouldBe " + tl);
+        wtu.checkCanvasRect(gl, halfWidth + quaterWidth, bottom, 2, 2, tr, "shouldBe " + tr, tolerance);
+        debug("Checking " + (flipY ? "bottom" : "top"));
+        wtu.checkCanvasRect(gl, quaterWidth, top, 2, 2, bl, "shouldBe " + bl);
+        wtu.checkCanvasRect(gl, halfWidth + quaterWidth, top, 2, 2, br, "shouldBe " + br, tolerance);
+    }
+}
+
+function runTestOnBindingTargetImageBitmap(bindingTarget, program, bitmaps, optionsVal,
+    internalFormat, pixelFormat, pixelType, gl, tiu, wtu)
+{
+    var cases = [
+        { sub: false, bitmap: bitmaps.noFlipYPremul, flipY: false, premultiply: true },
+        { sub: true, bitmap: bitmaps.noFlipYPremul, flipY: false, premultiply: true },
+        { sub: false, bitmap: bitmaps.noFlipYUnpremul, flipY: false, premultiply: false },
+        { sub: true, bitmap: bitmaps.noFlipYUnpremul, flipY: false, premultiply: false },
+        { sub: false, bitmap: bitmaps.flipYPremul, flipY: true, premultiply: true },
+        { sub: true, bitmap: bitmaps.flipYPremul, flipY: true, premultiply: true },
+        { sub: false, bitmap: bitmaps.flipYUnpremul, flipY: true, premultiply: false },
+        { sub: true, bitmap: bitmaps.flipYUnpremul, flipY: true, premultiply: false },
+    ];
+
+    for (var i in cases) {
+        runOneIterationImageBitmapTest(cases[i].sub, bindingTarget, program, cases[i].bitmap,
+            cases[i].flipY, cases[i].premultiply, optionsVal, internalFormat, pixelFormat, pixelType, gl, tiu, wtu);
+    }
+}
+
+function runImageBitmapTest(bitmaps, alphaVal, internalFormat, pixelFormat, pixelType, gl, tiu, wtu)
+{
+    var optionsVal = {alpha: alphaVal};
+    var program = tiu.setupTexturedQuad(gl, internalFormat);
+    runTestOnBindingTargetImageBitmap(gl.TEXTURE_2D, program, bitmaps, optionsVal,
+        internalFormat, pixelFormat, pixelType, gl, tiu, wtu);
+
+    // cube map texture must be square
+    if (bitmaps.noFlipYPremul.width == bitmaps.noFlipYPremul.height) {
+        program = tiu.setupTexturedQuadWithCubeMap(gl, internalFormat);
+        runTestOnBindingTargetImageBitmap(gl.TEXTURE_CUBE_MAP, program, bitmaps, optionsVal,
+            internalFormat, pixelFormat, pixelType, gl, tiu, wtu);
+    }
+
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "should be no errors");
+}
+
+function generateImageBitmap(source, alphaVal, internalFormat, pixelFormat, pixelType, gl, tiu, wtu)
```

Comment:
Could you please rename this to something like "runImageBitmapTest"? It has multiple callers, and it's confusing that they call both "createImageBitmap" and "generateImageBitmap".